### DISCUSSION TITLE
Csla blazor WebHost vs GenericHost

### DISCUSSION THREAD

**Discussion by rfcdejong** [MarimerLLC/csla#1959] [Ideas]
I was just wondering looking at the sample in here
https://github.com/MarimerLLC/csla/blob/master/Samples/BlazorExample/BlazorExample/Server/Program.cs

It is using IWebHost as written by Microsoft

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host
_This article covers the Web Host, which remains available only for backward compatibility. The Generic Host is recommended for all app types._

It feels better to me to update the sample to IHostBuilder

**Comment by rockfordlhotka** [comment]
You are probably correct, though ultimately I have the samples conform to the Microsoft project/solution templates.

What I've done with most of the Blazor apps (like the ones for my book) is to replace the project file(s) with ones created using the latest .NET 5 templates. For that _particular_ sample I did a manual upgrade to see what was required. I am not 100% happy with the result, 

As a result, I do plan to update that solution to .NET 5, replacing the existing projects with brand new ones created using the .NET 5 templates.

**Reply by rfcdejong** [reply]
Sounds good, new .NET 5 templates.

Microsoft moves fast lately, or during the time I was doing other work, all the .NET 5 stuff feels so new. 

Offtopic: I'm happy into being able to research my idea of using our existing business objects in a new Blazor project. It will be a huge 'migration', I hope I get things working. Currently, the usage of log4net LogManager is not usable, which we use extensively. Not CSLA related..

**Reply by rockfordlhotka** [reply]
If you are already using MobileFormatter for serialization then the move to Blazor should be pretty straightforward, from a CSLA perspective anyway.

The other big catch is that Microsoft now only supports `ClaimsPrincipal` and `ClaimsIdentity`, so custom principal/identity types are no longer a possibility.

Otherwise all the CSLA features - business rules, authz rules, async rules, etc. should all work as normal.
